consensus/scp/tests/mock_network/mod.rs

```diff
@@ -25,6 +29,37 @@ use std::{
     time::{Duration, Instant},
 };
 
+/// Values can be submitted to all nodes in parallel (true) or to nodes in sequential order (false)
+const SUBMIT_VALUES_IN_PARALLEL: bool = true;
+
+/// Total number of values to submit. Tests run until all values are externalized by all nodes.
+const VALUES_TO_PUSH: u32 = 200;
+
+/// Approximate rate that values are submitted to nodes.
+const VALUES_PER_SEC: u64 = 2000;
+
+/// We allow only a single proposal per slot, with up to this many values.
+const MAX_VALUES_PER_SLOT: u32 = 50;
+
+/// The total allowed testing time
+const MAX_TEST_TIME_SEC: u64 = 200;
+
+/// wait this long for slog to flush values
+const LOG_FLUSH_DELAY_MILLIS: u64 = 500;
+
+/// This parameter sets the interval for round and ballot timeout.
+/// SCP suggests one second, but threads can run much faster.
+const SCP_TIMEBASE: Duration = Duration::from_millis(100);
+
+/// Sleep when advancing to a new slot to help threads keep pace.
+const DURATION_TO_SLEEP_WHEN_ADVANCING_SLOT: Duration = Duration::from_millis(20);
+
+/// Because thread testing doesn't implement catchup, increase the lrucache used to store msgs.
+const OVERRIDE_LAST_SEEN_HISTORY_SIZE: usize = 100000;
```

Comment:
if one node falls far behind (perhaps because it's thread doesn't get any cpu time) it may be many slots behind its peers. When it finally gets some computation time, it needs a bunch of really old messages to catch up.

Reply:
That does not make sense to me - if it gets no CPU time, then it won't be processing messages and they would be stored in [this](https://github.com/mobilecoinofficial/mobilecoin/blob/e590421f61dda25cba91f3ddf676377a968dfe76/consensus/scp/tests/mock_network/mod.rs#L291) queue...

Reply:
Adding to it: the node is going to process messages in the order it received them, so it doesn't seem like it would need to have a history of slots to do its job. The last seen history prevents processing the same message twice and having a smaller history should not prevent nodes from doing their job, it is an optimization.